stdlib/public/SDK/Foundation/TypePreservingNSNumber.mm

```diff
@@ -23,26 +25,36 @@ @interface _SwiftTypePreservingNSNumber : NSNumber
 
 // This enum has a matching counterpart in Foundation.swift, please
 // update both copies when changing it.
-enum _SwiftTypePreservingNSNumberTag {
-  SwiftInt = 1,
-  SwiftUInt = 2,
-  SwiftFloat = 3,
-  SwiftDouble = 4,
-  SwiftCGFloat = 5,
-  SwiftBool = 6
+enum _SwiftTypePreservingNSNumberTag: uint8_t {
+  SwiftInt     =  0,
+  SwiftInt64   =  1,
+  SwiftInt32   =  2,
+  SwiftInt16   =  3,
+  SwiftInt8    =  4,
+  SwiftUInt    =  5,
+  SwiftUInt64  =  6,
+  SwiftUInt32  =  7,
+  SwiftUInt16  =  8,
+  SwiftUInt8   =  9,
+  SwiftFloat   = 10,
+  SwiftDouble  = 11,
+  SwiftCGFloat = 12,
+  SwiftBool    = 13,
+  NonSwift     = 14, // a non-type-preserving NSNumber from Cocoa
 };
 
+static const unsigned StorageSize = 8;
+
 @implementation _SwiftTypePreservingNSNumber {
+  /*alignas(StorageSize)*/ char storage[StorageSize];
```

Comment:
Unfortunately, Clang doesn't seem to let you put any `alignas` decorations at all in an ObjC++ implementation without an "attributes can't be put here" error.